apps/extension/src/ui/hooks/useResolveEnsName.ts

```diff
@@ -0,0 +1,35 @@
+import { isEthereumAddress } from "@polkadot/util-crypto"
+import { isPotentialEns } from "@talismn/on-chain-id"
+import {
+  ensNamesIsFetching,
+  readEnsNameLookups,
+  useResolveAddressesForEnsNames,
+} from "@ui/atoms/ensNames"
+import { useRecoilValue } from "recoil"
+
+export const useResolveEnsName = (name?: string) => {
```

Comment:
I may be missing something but this + ensNames.ts (mainly) feels overengineered and hard to read/maintain.
wdyt about using a useQuery call instead ?